app/models/customer.rb

```diff
@@ -0,0 +1,15 @@
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless value =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+      record.errors[attribute] << (options[:message] || "is not an email")
+    end
+  end
+end
```

Comment:
This is an interesting approach, didn't know these existed. But it seems a bit too much to create a new class (and if you do it probably should be located in a seperate file). I think that adding a format validation in your Customer class should be enough:
'validates_format_of :email,:with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i`

Reply:
I think so. I never used custom validators in a form of a seperate class. We can talk about it during our next meeting.